src/groupWith.js

```diff
@@ -0,0 +1,33 @@
+var _curry2 = require('./internal/_curry2');
+
+/**
+ * Takes a list and returns a list of lists where each sublist contains only equal elements.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig (a, a -> Boolean) -> [a] -> [[a]]
+ * @param {Function} fn Function :: a, a -> Boolean
+ * @param {Array} list The array to group
+ * @return {List} A list that contains sublists of equal elements,
+ *         whose concatenations is equal to the original list.
+ * @example
+ *
+ *      var groupWithLetters = R.groupWith(R.equals);
+ *      var letters = ["M", "i", "s", "s", "i", "s", "s", "i", "p", "p", "i"];
+ *      groupWithLetters(letters);
+ *      // [['M'], ['i'], ['s', 's'], ['i'], ['s', 's'], ['i'], ['p', 'p'], ['i']]
+ */
+module.exports = _curry2(function(fn, list) {
+  var result = [];
+  var idx = 0;
+  while (idx < list.length) {
+    var nextidx = idx + 1;
+    while (nextidx < list.length && fn(list[idx], list[nextidx])) {
+      nextidx += 1;
+    }
+    result.push(list.slice(idx, nextidx));
```

Comment:
Should we be using our internal `_slice` here?

If we did, of course, it would end up with what's in the documentation now, `[['a', 'e'], ['s', 't'], ['i', 'o', 'u']]` rather than what David points out it should now return of `['ae', 'st', 'iou']`.

The question is whether this should work on anything list-like or just lists (and strings).